ui/src/index.test.tsx

```diff
@@ -0,0 +1,167 @@
+// Globals
+
+const LEDGER_HOST = 'localhost';
+const LEDGER_PORT = '6865';
+const LEDGER_ID = "DAVL";
+const APPLICATION_ID = "DAVL";
+const SECRET = "secret";
+//const LEDGER_URL = 'http://localhost:7575/';
+const JSON_API_PORT = '7575';
+const UI_PORT = '3000';
+const TEST_SETUP = '../test-setup.json'
+const INIT_CMD = 'v5-init'
+const DARS = [
+    '../released/davl-v4.dar',
+    '../released/davl-v5.dar',
+    '../released/davl-upgrade-v4-v5.dar'
+]
+let ui: ChildProcess | undefined = undefined;
+let browser: Browser | undefined = undefined;
+let api: ChildProcess | undefined = undefined;
+let ledger: ChildProcess | undefined = undefined;
+
+////////////////////////////////////////////////////////////////////////////////
+// Setup/teardown
+
+/* eslint-disable @typescript-eslint/require-await */
+beforeAll(async () => {
+  // Create log files.
+  const [ledgerLog, apiLog, uiLog] =
+    [ tmp.fileSync({ prefix: 'sandbox-', postfix: '.log' }),
+      tmp.fileSync({ prefix: 'json-api-', postfix: '.log' }),
+      tmp.fileSync({ prefix: 'davl-ui-', postfix: '.log' }),
+    ];
+  console.log(`ledger log : "${ledgerLog.name}"`);
+  console.log(`json-api log : "${apiLog.name}"`);
+  console.log(`ui log : "${uiLog.name}"`);
+
+  // Spin up a ledger.
+  ledger =
+    spawn('daml',
+          [ 'sandbox', '--port', LEDGER_PORT, '--ledgerid', LEDGER_ID, '--wall-clock-time' ],
+          { detached: true, stdio: ['ignore', ledgerLog.fd, ledgerLog.fd ] }
+         );
+  ledger.unref();
+  await waitOn({resources: [ `tcp:localhost:${LEDGER_PORT}` ]});
+
+  // Upload dars (sync).
+  for (const dar of DARS) {
+    const {status} =
+      spawnSync('daml',
+        [ 'ledger', 'upload-dar', '--host', LEDGER_HOST, '--port', LEDGER_PORT, dar ],
+        { stdio: [ 'ignore', 'inherit', 'inherit' ] });
+    if(status !== undefined && status !== 0) {
+      exit(1);
+    }
+  }
+
+  // Spint up a json-api.
+  api = spawn('daml',
+              ['json-api', '--ledger-host', LEDGER_HOST, '--ledger-port', LEDGER_PORT, '--http-port', JSON_API_PORT],
+              { detached: true, stdio: ['ignore', apiLog.fd, apiLog.fd ] });
+  api.unref();
+  await waitOn({resources: [ `tcp:localhost:${JSON_API_PORT}` ]});
+
+  // Populate initial ledger state (sync).
+  const init =
+     spawnSync ('yarn',
+            ['run', 'davl-admin-cli', INIT_CMD, '-f', TEST_SETUP],
+            { stdio: [ 'ignore', 'inherit', 'inherit' ] }
+           );
+  if(init.status !== undefined && init.status !== 0) {
+    exit(1);
+  }
+
+  // Spin up a UI.
+  ui = spawn('yarn',
+             ['start'],
+             { detached: true, env: {...process.env, BROWSER: 'none'}, stdio: ['inherit', uiLog.fd, uiLog.fd ]});
+  ui.unref();
+  await waitOn({resources: [ `tcp:localhost:${UI_PORT}` ]});
+
+  // Launch a browser.
+  browser = await puppeteer.launch();
+
+}, 40_000);
+
+/* eslint-disable @typescript-eslint/require-await */
+afterAll(async () => {
+  if (ledger) {
```

Comment:
```suggestion
  if (ledger !== undefined) {
```
Truthiness and falsyness is a misfeature IMO.